src/app/util/attribute-storage.cpp

```diff
@@ -1080,6 +1093,21 @@ CHIP_ERROR GetSemanticTagForEndpointAtIndex(EndpointId endpoint, size_t index,
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR GetEndpointUniqueIdForEndPoint(EndpointId endpoint, MutableCharSpan & epUniqueIdMutSpan)
+{
+    uint16_t endpointIndex = emberAfIndexFromEndpoint(endpoint);
+
+    if (endpointIndex == 0xFFFF)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    CharSpan epUniqueIdSpan(emAfEndpoints[endpointIndex].endpointUniqueId, emAfEndpoints[endpointIndex].endpointUniqueIdSize);
+    CopyCharSpanToMutableCharSpan(epUniqueIdSpan, epUniqueIdMutSpan);
```

Comment:
```suggestion
    return CopyCharSpanToMutableCharSpan(epUniqueIdSpan, epUniqueIdMutSpan);
```

Otherwise this happily returns CHIP_NO_ERROR on failure to copy!

Reply:
Agree.